packages/payload/src/folders/hooks/ensureSafeCollectionsChange.ts

```diff
@@ -0,0 +1,143 @@
+import { APIError, type CollectionBeforeValidateHook, type CollectionSlug } from '../../index.js'
+import { extractID } from '../../utilities/extractID.js'
+import { getTranslatedLabel } from '../../utilities/getTranslatedLabel.js'
+
+export const ensureSafeCollectionsChange =
+  ({ foldersSlug }: { foldersSlug: CollectionSlug }): CollectionBeforeValidateHook =>
+  async ({ data, originalDoc, req }) => {
+    const currentFolderID = extractID(originalDoc || {})
+    const parentFolderID = extractID(data?.folder || originalDoc?.folder || {})
+    if (Array.isArray(data?.folderType) && data.folderType.length > 0) {
+      const folderType = data.folderType as string[]
+      const currentlyAssignedCollections: string[] | undefined =
+        Array.isArray(originalDoc?.folderType) && originalDoc.folderType.length > 0
+          ? originalDoc.folderType
+          : undefined
+      /**
+       * Check if the assigned collections have changed.
+       * example:
+       * - originalAssignedCollections: ['posts', 'pages']
+       * - folderType: ['posts']
+       *
+       * The user is narrowing the types of documents that can be associated with this folder.
+       * If the user is only expanding the types of documents that can be associated with this folder,
+       * we do not need to do anything.
+       */
+      const newCollections = currentlyAssignedCollections
+        ? // user is narrowing the current scope of the folder
+          currentlyAssignedCollections.filter((c) => !folderType.includes(c))
+        : // user is adding a scope to the folder
+          folderType
+
+      if (newCollections && newCollections.length > 0) {
+        let hasDependentDocuments = false
+        if (typeof currentFolderID === 'string' || typeof currentFolderID === 'number') {
+          const childDocumentsResult = await req.payload.findByID({
+            id: currentFolderID,
+            collection: foldersSlug,
+            joins: {
+              documentsAndFolders: {
+                limit: 100_000_000,
+                where: {
+                  or: [
+                    {
+                      relationTo: {
+                        in: newCollections,
+                      },
+                    },
+                  ],
+                },
+              },
+            },
+            overrideAccess: true,
+            req,
+          })
+
+          hasDependentDocuments = childDocumentsResult.documentsAndFolders.docs.length > 0
+        }
+
+        // matches folders that are directly related to the removed collections
+        let hasDependentFolders = false
+        if (
+          !hasDependentDocuments &&
+          (typeof currentFolderID === 'string' || typeof currentFolderID === 'number')
+        ) {
+          const childFoldersResult = await req.payload.find({
+            collection: foldersSlug,
+            limit: 1,
```

Comment:
Should this have the `req` in case the child folder needs to be found within the same transaction as a create/update?

Reply:
Implemented https://github.com/payloadcms/payload/pull/12797/commits/03b50b1c981ace36b9c232245b470078696c801d